src/main/java/com/iota/iri/utils/log/interval/IntervalLogger.java

```diff
@@ -0,0 +1,490 @@
+            lastReceivedMessage = newMessage;
+
+            triggerOutput();
+        }
+
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * It checks if the given message is new and then triggers the output.
+     */
+    @Override
+    public IntervalLogger debug(String message) {
+        Message newMessage = new DebugMessage(message);
+
+        if (!newMessage.equals(lastReceivedMessage)) {
+            lastReceivedMessage = newMessage;
+
+            triggerOutput();
+        }
+
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * It checks if the given message is new and then triggers the output.
+     *
+     * Error messages will always get dumped immediately instead of scheduling their output. Since error messages are
+     * usually a sign of some misbehaviour of the node we want to be able to see them when they appear (to be able to
+     * track down bugs more easily).
+     */
+    @Override
+    public IntervalLogger error(String message) {
+        return error(message, null);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * It checks if the given message is new and then triggers the output.
+     *
+     * Error messages will always get dumped immediately instead of scheduling their output. Since error messages are
+     * usually a sign of some misbehaviour of the node we want to be able to see them when they appear (to be able to
+     * track down bugs more easily).
+     */
+    @Override
+    public IntervalLogger error(String message, Throwable cause) {
+        Message newMessage = new ErrorMessage(message, cause);
+
+        if (!newMessage.equals(lastReceivedMessage)) {
+            lastReceivedMessage = newMessage;
+
+            triggerOutput(true);
+        }
+
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean getEnabled() {
+        return enabled;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IntervalLogger setEnabled(boolean enabled) {
+        this.enabled = enabled;
+
+        return this;
+    }
+
+    /**
+     * Does the same as {@link #triggerOutput(boolean)} but defaults to letting the logger decide when to print the
+     * message.
+     */
+    public void triggerOutput() {
+        triggerOutput(false);
+    }
+
+    /**
+     * Triggers the output of the last received message.
+     *
+     * It either prints the message immediately (if enough time has passed since the last output or requested by the
+     * caller) or schedules it for the next interval.
+     *
+     * @param printImmediately flag indicating if the messages should be scheduled or printed immediately
+     */
+    public void triggerOutput(boolean printImmediately) {
+        if (lastReceivedMessage != null && (printImmediately ||
+                System.currentTimeMillis() - lastLogTime >= logInterval)) {
+
+            lastReceivedMessage.print();
+
+            if (outputThread != null) {
```

Comment:
Because it gets nullified by the thread and is used as a flag to know if we have a thread already.

See other comments about requiring a closure rather than a single running thread to reduce the resource requirements in case we only rarely post messages to this logger.